GLCore Fixes

TVsIan opened the pull request:
A few changes to improve GLCore & shader selection.

- GLCore will only activate if the OpenGL version is 3.1 or higher and the GPU is nVidia or AMD. Intel iGPU performance is spotty on GLCore, but it will still be available as a manual option. This may need to be revisited once Intel starts releasing dedicated cards.
- Mega Bezel shaders will only appear in the list for x86/x86_64. The shaders will still be present for manual selection, but as they won't work on most (any?) other systems, they won't be cluttering the list.

The shader list change was a minor tweak to how the list of shaders is generated in batocera-shaders, it should be self-explanatory if new options are added or options are removed. This could also be further customized for other platforms that may need even fewer options due to very weak GPUs.

nadenislamarre commented:
ok.
can we use this pr to do one point :
create a function getGlVersionAndVendor() into utils/videoMode.py for example to remove this code from lr ?

TVsIan commented:
Ok, I added two functions to utils/videoMode.py - getGLVersion() and getGLVendor(). That way they can be used independently, just in case. It's been removed from the Libretro generator and replaced with links to the new functions.

nadenislamarre commented:
and while videoMode.getGLVersion() >= 3.1 and videoMode.getGLVendor()  are exec call, there takes some time,
you should call them only once at the begining of generate()
and pass the 2 values to writeLibretroConfig to save 0.2s at each game boot.

TVsIan commented:
Will do, I'll knock that out.

On Thu, Jan 6, 2022 at 2:27 AM Nicolas Adenis-Lamarre <
***@***.***> wrote:

> and while videoMode.getGLVersion() >= 3.1 and videoMode.getGLVendor() are
> exec call, there takes some time,
> you should call them only once at the begining of generate()
> and pass the 2 values to writeLibretroConfig to save 0.2s at each game
> boot.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/batocera-linux/batocera.linux/pull/5346#issuecomment-1006342244>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE5KD2WLO5EP5GJ7X5IKWG3UUU77TANCNFSM5LJAI25A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

TVsIan commented:
We should be ready to go. For these final commits:

- Moved backend detection/selection to a single function. The shell commands are only called if the backend is left on auto.
- Added the lists of cores that will force gl or glcore (some of which are not currently in Batocera), and will flip the result if needed in order to pick the proper shaders.
- Added a Retroarch option in ES to disable the "Allow Core Rotation" option. This is mostly for handhelds like the Lynx and Wonder Swan but may also affect arcade cores. Turning it off will fix Mega Bezel-based vertical shaders as well as vertical games for users with a rotatable monitor.

Tested a few combinations of settings and cores and everything appears to be working as intended.
